homeassistant/components/modem_callerid/sensor.py

```diff
@@ -84,38 +101,53 @@ def name(self):
         """Return the name of the sensor."""
         return self._name
 
+    @property
+    def unique_id(self):
+        """Return the unique id of the sensor."""
+        return f"{self._server_unique_id}/{self._name}"
+
     @property
     def extra_state_attributes(self):
         """Return the state attributes."""
         return self._attributes
 
-    def _stop_modem(self, event):
-        """HA is shutting down, close modem port."""
-        if self.modem:
-            self.modem.close()
-            self.modem = None
-
     def _incomingcallcallback(self, newstate):
         """Handle new states."""
-        if newstate == self.modem.STATE_RING:
-            if self.state == self.modem.STATE_IDLE:
+        if newstate == self.api.STATE_RING:
+            if self.state == self.api.STATE_IDLE:
                 att = {
-                    "cid_time": self.modem.get_cidtime,
+                    "cid_time": self.api.get_cidtime,
                     "cid_number": "",
                     "cid_name": "",
                 }
                 self.set_attributes(att)
             self._state = STATE_RING
             self.schedule_update_ha_state()
-        elif newstate == self.modem.STATE_CALLERID:
+        elif newstate == self.api.STATE_CALLERID:
             att = {
-                "cid_time": self.modem.get_cidtime,
-                "cid_number": self.modem.get_cidnumber,
-                "cid_name": self.modem.get_cidname,
+                "cid_time": self.api.get_cidtime,
+                "cid_number": self.api.get_cidnumber,
+                "cid_name": self.api.get_cidname,
             }
             self.set_attributes(att)
             self._state = STATE_CALLERID
             self.schedule_update_ha_state()
-        elif newstate == self.modem.STATE_IDLE:
+        elif newstate == self.api.STATE_IDLE:
             self._state = STATE_IDLE
             self.schedule_update_ha_state()
+
+    def reject_call(self) -> None:
+        """Reject Incoming Call."""
+        self.api.reject_call(self.device)
+
+
+def setup(hass):
+    """Shutdown modem with Home Assistant restart."""
+    hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP, _stop_modem)
+
+
+def _stop_modem(self, event):
+    """HA is shutting down, close modem port."""
+    if self.api:
+        self.api.close()
+        self.api = None
```

Comment:
Looks like this is never called.  If it is needed, be sure to cancel `hass.bus.listen_once` with the callable it returns when unloading the integration.

If you can call it from async context you can do

`config_entry.async_on_unload(ass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, _stop_modem))`

Reply:
Perhaps we don't need it at all. My testing so far shows that removing and re-adding the integration still haas the modem working.